File: python/easy_two_sum.py
class Solution():
    
  def two_sum(self, total_number: list[int], target: int) -> list[int]:
   previous_values = []

   for index in range(1,len(total_number)):
      current_number = total_number[index]
      needed_value = target - current_number
      index2 = previous_values[needed_value]

   if index2 != 0:
      return [index2, index]
   else: 
      previous_values[current_number] = index

if __name__ == "__main__":
   sm = Solution()
   array_numbers = [2,7,11,15]
   target_numbers = 9

   result = sm.two_sum(array_numbers, target_numbers) # => 0,1

   if result != 0:
      print('Sum is ', result)
   else:
      print('Numbers not found to sum') 

File: typescript/algorithms/easy-two-sum.ts
/**
* GET INDEX OF NUMBER WHERE TWO NUMBER FOR ARRAY EQUAL GOAL
* @param totalNumber {number[]} totalNumber {Number[]}
* @param target {number} target Number
* @returns {number[]} Result

arr = [2,3,5,6,7,8,9,10]
b(o)2
result = 8.

*/

const twoSum = (
	totalNumber: number[],
	target: number
): number[] | undefined => {
	let previousValues: any = [];

	for (let index = 0; index < totalNumber.length; index++) {
		let currentNumber: number = totalNumber[index];
		let neededValue: number = target - currentNumber;
		let index2 = previousValues[neededValue];

		if (index2 != undefined) {
			return [index2, index];
		} else {
			previousValues[currentNumber] = index;
		}
	}
};

const array_numbers: any[] = [2, 7, 11, 15];
const target_numbers: number = 9;

console.log(twoSum(array_numbers, target_numbers));
